raft/server.cc

```diff
@@ -454,6 +462,47 @@ future<> server_impl::wait_for_state_change(seastar::abort_source* as) {
     }
 }
 
+future<> server_impl::trigger_snapshot(seastar::abort_source& as) {
```

Comment:
All other functions make abort_source optional. I know you are against default parameters, but optional parameter does not have to have a default, so a user of the API has to provide null explicitly if he does not want it to be abortable.

Reply:
It will complicate the implementation and for what? To make it easier for the user to shoot themselves in the foot?

Reply:
To make API consistent. Would it complicate it too much?

Reply:
Not much, just sprinkle a bunch of `if`s and `optional`s. But I'd rather make the API consistent in the other direction.

Reply:
> But I'd rather make the API consistent in the other direction.

But we will not so it in this patch series.